pennylane/grouping/__init__.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This subpackage defines functions and classes for Pauli word partitioning
```

Comment:
```suggestion
This subpackage defines functions and classes for Pauli-word partitioning
```

Reply:
Just double-checking if we'd like to include a dash here (it'd be recurring throughout the codebase also next to `grouping`). A few sources include it without ([here](https://arxiv.org/pdf/1908.05628.pdf) and [here](https://authors.library.caltech.edu/103906/2/1908.08067.pdf)),  but could transition to having it with a dash.

Reply:
use a dash whereever the phrase "Pauli-word" is used as an adjective (e.g., "Pauli-word partitioning", and without a dash when it's a standalone phrase)

Reply:
unfortunately, this trips up even native English speakers all the time; I try to correct it when I see it

Reply:
Oh, that's something really new indeed, thanks for clarifying it! :blush: